WordPress/src/main/java/org/wordpress/android/modules/ViewModelFactory.java

```diff
@@ -0,0 +1,43 @@
+package org.wordpress.android.modules;
+
+import android.arch.lifecycle.ViewModel;
+import android.arch.lifecycle.ViewModelProvider;
+import android.support.annotation.NonNull;
+import android.support.v4.util.ArrayMap;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+@Singleton
+public class ViewModelFactory implements ViewModelProvider.Factory {
+
+    private final ArrayMap<Class<? extends ViewModel>, Provider<ViewModel>> mViewModelsMap;
+
+    @Inject
+    public ViewModelFactory(ArrayMap<Class<? extends ViewModel>, Provider<ViewModel>> viewModelsMap) {
+        this.mViewModelsMap = viewModelsMap;
+    }
+
+    @NonNull
```

Comment:
We can add the `@SuppressWarnings("unchecked")` here to avoid the AS warnings.

Reply:
From an unknown reason, I don't see the warning unless I run the inspection manually:(.